ocaml-lsp-server/docs/ocamllsp/hoverExtended-spec.md

```diff
@@ -0,0 +1,52 @@
+#### Hover Extended
+
+Alternative hover command providing additional information.
+
+This command is has support for variable verbosity.
+
+```ocaml
+type t = int
+let x : t = 1
+```
+
+With the cursor on the value `x`, a call with a verbosity of 0 or lower would return `t` making it equivalent to a call to `textDocument/hover`. A call with a verbosity of 1 would return `int`.
+
+The management of the verbosity is left to the clients to implement.
+
+##### Client capability
+
+nothing that should be noted
+
+##### Server capability
+
+property name: `handleHoverExtended`
+property type: `boolean`
+
+##### Request
+
+- method: `ocamllsp/hoverExtended`
+- params:
+
+  ```json
+  {
+    "textDocument": TextDocumentIdentifier,
+    "position": Position,
+    "verbosity": integer
+  }
+  ```
+
+##### Response
+
+The response is similar to the one of `textDocument/hover`
+
+```typescript
+	/**
+	 * The hover's content
+	 */
+	contents: MarkedString | MarkedString[] | MarkupContent;
+	/**
+	 * An optional range is a range inside a text document
+	 * that is used to visualize a hover, e.g. by changing the background color.
+	 */
+	range?: Range;
+```
```

Comment:
the verbosity could also be returned. My idea is that a client could rely on the server to handle the verbosity most of the time, not passing it in the request. While storing the returned verbosity. Then it could offer some additional commands like `hoverLessVerbose` (to send the stored verbosity minus one) or `hoverSameVerbose` (to request the same verbosity)